File: VB/02_Text/EmbedFontToPdf.vb
Imports Spire.Pdf.Conversion

Namespace EmbedFontToPdf
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfEmbeddedFontConverter object with the path to an existing PDF file
			Dim converter As New PdfEmbeddedFontConverter("..\..\..\..\..\..\Data\UnEmbed.pdf")

			' Specify the output file name for the converted PDF document
			Dim output As String = "EmbedFontToPdf.pdf"

			' Convert the input PDF by embedding the fonts and save it as a new document
			converter.ToEmbeddedFontDocument(output)

			' Dispose the converter
			converter.Dispose()

			' Launch the Pdf file
			PDFDocumentViewer(output)
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/02_Text/EmbedFontToPdf.cs
using System;
using System.Drawing;
using System.IO;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Annotations;
using Spire.Pdf.ColorSpace;
using Spire.Pdf.Conversion;
using Spire.Pdf.Graphics;


namespace EmbedFontToPdf
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {           
            // Creating an instance of the PdfEmbeddedFontConverter class and specifying the path to the source PDF file
            PdfEmbeddedFontConverter converter = new PdfEmbeddedFontConverter(@"..\..\..\..\..\..\Data\UnEmbed.pdf");

            // Specifying the output file path where the converted PDF with embedded fonts will be saved
            string output = @"EmbedFontToPdf.pdf";

            // Calling the ToEmbeddedFontDocument method of the converter object to convert the PDF and embed fonts
            converter.ToEmbeddedFontDocument(output);

            //Launch the Pdf file
            PDFDocumentViewer(output);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
